read.go

```diff
@@ -269,10 +265,52 @@ func readNativeFrames(d dicomio.Reader, parsedData *Dataset, fc chan<- *frame.Fr
 
 	debug.Logf("readNativeFrames:\nRows: %d\nCols:%d\nFrames::%d\nBitsAlloc:%d\nSamplesPerPixel:%d", MustGetInts(rows.Value)[0], MustGetInts(cols.Value)[0], nFrames, bitsAllocated, samplesPerPixel)
 
+	bytesAllocated := bitsAllocated / 8
+	bytesToRead = bytesAllocated * samplesPerPixel * pixelsPerFrame * nFrames
+	if bitsAllocated == 1 {
+		bytesToRead = pixelsPerFrame * samplesPerPixel / 8 * nFrames
+	}
+
+	skipOne := false
+	switch {
+	case uint32(bytesToRead) == vl: // we good nothing
+	case uint32(bytesToRead) == vl-1 && vl%2 == 0:
+		skipOne = true
+	case uint32(bytesToRead) == vl-1 && vl%2 != 0:
+		return nil, 0, fmt.Errorf("vl=%d: %w", vl, ErrorExpectedEvenLength)
+	default:
```

Comment:
Looking at this again, I don't love the readability of this block here. I'd like something that makes it clear to the reader that this block is specifically handling the special cases for when `bytesToRead != vl` and for that to be very clear. Needing to have the "do nothing" case further makes this pattern feel weird to me.

Even though this has an additional level of nesting, I wonder if the following is ultimately more readable:

```go
if bytesToRead != vl {
    // Block of code handling all situations where bytesToRead != vl.
    switch {
    bytesToRead == vl-1 && vl%2 == 0:
        skipFinalPaddingByte = true
    bytesToRead == vl-1 && vl%2 !=0:
        return nil, 0, fmt.Errorf("vl=%d: %w", vl, ErrorExpectedEvenLength)
    default:
        // bytesToRead != vl and none of the special conditions are met, so this is an error or fallback behavior should be triggered
        ....
    }
}
```